spec/mongo/transactions_examples_spec.rb

```diff
@@ -0,0 +1,236 @@
+require 'spec_helper'
+
+describe 'transactions examples in Ruby', if: test_transactions? do
+
+  let(:hr) do
+    authorized_client.use(:hr).database
+  end
+
+  let(:reporting) do
+    authorized_client.use(:reporting).database
+  end
+
+  before(:each) do
+    hr[:employees].insert_one(employee: 3, status: 'Active')
+    reporting[:events].insert_one(employee: 3, status: { new: 'Active', old: nil})
+  end
+
+  after(:each) do
+    hr.drop
+    reporting.drop
+  end
+
+  context 'individual examples' do
+
+    let(:session) do
+      authorized_client.start_session
+    end
+
+    # Start Transactions Intro Example 1
+
+    def update_employee_info(session)
+      employees_coll = session.client.use(:hr)[:employees]
+      events_coll = session.client.use(:reporting)[:events]
+
+      session.start_transaction(read_concern: { level: :snapshot },
+                                write_concern: { w: :majority })
+      employees_coll.update_one({ employee: 3 }, { '$set' => { status: 'Inactive'} },
```

Comment:
Is `{ '$set' => { status: 'Inactive'} }` equivalent to `{ status: 'Inactive'}` ?

Reply:
Technically, `update_one` is not allowed to be called without a `$set` (or some other `$` key first); `replace_one` is supposed to be used for cases when the entire document is overwritten. I followed what the Python driver did here.

Reply:
Gotcha.